java/vector/src/main/java/org/apache/arrow/vector/IntVector.java

```diff
@@ -164,7 +165,9 @@ private void setValue(int index, int value) {
    * @param value value of element
    */
   public void set(int index, int value) {
-    BitVectorHelper.setBit(validityBuffer, index);
+    if (!NON_NULLABLE_VECTORS_ENABLED || nullable) {
```

Comment:
Can we declare `NON_NULLABLE_VECTORS_ENABLED && nullable` as a final instance variable?  If so, we can avoid a similar conditional expression.

Reply:
@kiszk Thanks for your feedback. I have changed the code accordingly. 
It really makes the code simpler, and I've verified that there were no performance regression. 

However, it could also lead to some consufing behavior. For example, a vector that is known to have no null values may have `nullable` field equal to `true`.